### DISCUSSION TITLE
New in v9.6.0 - ✨ Improved Chat UI - Now showing what's been shared with the LLM

### DISCUSSION THREAD

**Discussion by olimorris** [olimorris/codecompanion.nvim#429] [Announcements]
https://github.com/user-attachments/assets/7828d866-9f11-4aa4-ae56-8f1a1e0cdf30

A common ask (#395 #311) has been to visually show the user what's been shared with the LLM. Copilot Chat in VS Code has a really nice way of showing references and so I've built that into the plugin for this release.

Behind the scenes, a references table in the Chat class now links references to messages via an ID. This will make it very easy for you to be able to remove references directly in the chat buffer by deleting them. This will be coming soon. The references table will also make it easier to allow for things like "sticky" files and buffers...that is, content that self-updates with every response. It's cumbersome to have to continually update the content manually, especially if you're using the `@editor` tool.

Hope you enjoy this release!

**Comment by GitMurf** [comment]
> This will make it very easy for you to be able to remove references directly in the chat buffer by deleting them. This will be coming soon.

@olimorris is the ability to remove references available yet? I haven't stumbled upon it yet in the announcements or the code but wanted to make sure I hadn't missed something. Thanks!

**Reply by olimorris** [reply]
Not yet. Hoping that will be sometime this week and I'll definitely announce it. Will probably add omnifunc stuff first
